src/diffusers/pipelines/stable_diffusion_xl/pipeline_stable_diffusion_xl.py

```diff
@@ -1172,6 +1172,10 @@ def __call__(
                 self.do_classifier_free_guidance,
             )
 
+        # expand the image embeddings if we are using perturbed-attention guidance
+        for i in range(len(image_embeds)):
+            image_embeds[i] = image_embeds[i].repeat(prompt_embeds.shape[0] // latents.shape[0], 1, 1)
```

Comment:
This throws an error with the PLUS versions of IP Adapters each `image_embeds` is a 4D tensor with the PLUS versions.

Reply:
Thank you for finding the error! I found the cause of the error and, thanks to this, came up with a more elegant design. I will upload the revised code with the results soon!